syntax = "proto2";

import "player.proto";
option java_package = "proto";
option java_outer_classname = "UdpPacketProtos";

message Scores {
  required string name = 1;
  required int32 score = 2;
}

message UdpPacket {
  enum PacketType {
    DRAW = 0;
    WINNER = 1;
    TURN = 2;
    TIME = 3;
    TIMEOUT = 4;
    PORT = 5;
    ENDGAME = 6;
  }

  required PacketType type = 1;

  message DrawPacket {
    required PacketType type = 1;
    optional int32 x = 2;
    optional int32 y = 3;
    optional string color = 4;
    optional int32 width = 5;
    optional bool start = 6;
    optional bool clear = 7;
  }

  message WinnerPacket {
    required PacketType type = 1;
    required Player player = 2;
  }

  message TimeoutPacket {
    required PacketType type = 1;
  }

  message TurnPacket {
    required PacketType type = 1;
    required Player player = 2;
    repeated Scores scores = 3;
    optional string object = 4;
  }

  message TimePacket {
    required PacketType type = 1;
    required int32 time = 2;
  }

  message PortPacket {
    required PacketType type = 1;
  }

  message EndgamePacket {
    required PacketType type = 1;
    required string winner = 2;
  }
}
